### PR TITLE
fix(medusa): Query SalesChannel Products in storefront

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**
- Adds the ability to query products in a sales channel from a storefront
- Adds feature flag decorator to query param `sales_channel_id` in `GET /admin/products` as the feature is still guarded

**Tests**
- Retrieve products in specific channel
- Retrieve products in multiple channels
- Retrieve all products when no channel is defined

Fixes CORE-602.

### CODE REVIEW

**Review by adrien2p — APPROVED** [review]
LGTM

File: packages/medusa/src/api/routes/store/products/list-products.ts [hunk 1]
```diff
@@ -1,28 +1,30 @@
-import {
```
[the remark this answers is not in the document]
Comment (adrien2p) [reply packages/medusa/src/api/routes/store/products/list-products.ts:1]: Just so that each time we touch and handler we can refactor it to be aligned

**Review by srindom — APPROVED** [review]
LGTM

File: packages/medusa/src/api/routes/store/products/list-products.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (olivermrbl) [reply packages/medusa/src/api/routes/store/products/list-products.ts:1]: As this PR does not touch the handler's functionality, I'd rather leave it for a separate one, if that's cool with you? I'd like to get this merged asap.

[the remark this answers is not in the document]
Comment (adrien2p) [reply packages/medusa/src/api/routes/store/products/list-products.ts:1]: yeah sure, it was just a thoughts but hey

**Comment by srindom** [comment]
@olivermrbl - maybe add a changeset?

**Comment by olivermrbl** [comment]
> maybe add a changeset?

PRs that include features guarded by feature flags generally should not have a changeset

**Pull request merged**